packages/api/src/hooks/useDocumentUpload.ts

```diff
@@ -1,19 +1,78 @@
-import { useCallback, useMemo } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import useMutation from '../useMutation';
+import { compressImageFile, generateChunks, numToUint8Array, combineBuffer, readFile } from '../utils';
+import md5 from 'md5';
+import { getWebsocket } from '../APIProvider';
 
 type TDocumentUploadPayload = Parameters<ReturnType<typeof useMutation<'document_upload'>>['mutate']>[0]['payload'];
+type TMutatePayload = TDocumentUploadPayload & {
+    file?: File;
+};
 
 /** A custom hook to handle document file uploads to our backend. */
 const useDocumentUpload = () => {
     const { data, mutate, ...rest } = useMutation('document_upload');
+    const [tempChunks, setTempChunks] = useState<Uint8Array[]>([]);
+    const derivAPI = getWebsocket();
```

Comment:
```suggestion
    const activeWebsocket = getWebsocket();
```
actually, this is an active `WebSocket` instance not `DerivAPI`. Dont wanna be sus in naming things